src/app/(admin)/admin/settings/rooms/[id]/page.tsx

```diff
@@ -1,15 +1,50 @@
+"use client";
 import AdminButton from "~/app/_components/admin/Button";
+import BackButton from "../../../tasters/@show/_backButton";
 import React from "react";
 import Link from "next/link";
+import { api } from "~/trpc/react";
+import { useRouter } from "next/navigation";
 
-const page = () => {
+import { InfoCard } from "~/app/_components/admin/InfoCard";
+import LoadingSpinner from "~/app/_components/admin/LoadingSpinner";
+
+const ShowRoom = ({  params: { id }  }: { params: { id: string } }) => {
+    const { data, isLoading, isError, error } = api.rooms.show.useQuery({ id });
+    const router = useRouter();
+
+  
+    if (isLoading) {
+      return <div>Loading...</div>;
+    }
+    if (isError) return <p>{error.message}</p>;
     return (
-        <div><p>Hello</p>
-            <AdminButton>
-            <Link href="/admin/settings/rooms/update">Update this room</Link>
-            </AdminButton>
+        <div className="h-full w-full p-2">
+          <span className="flex justify-between">
+            <h3 className="text-xl font-bold">Room</h3>
+            <span className="flex gap-2">
+              <AdminButton>
+                <Link href="/admin/settings/rooms/update">✏️ Edit</Link>
+              </AdminButton>
+          
+              <BackButton
+                returnPath="/admin/settings/rooms"
+                label="⬅️ Back to list"
+              />
+            </span>
+          </span>
+          <article className="grid grid-cols-1 gap-6 p-2 sm:grid-cols-2">
+            <InfoCard 
+              title={data.name}
```

Comment:
I have "data" underscored in red in the VS code, could you please double check on that?

Reply:
Just had a look. UI looks great. Red line is because `api.rooms.show` can return either an object or undefined. If it's undefined then we can't do `data.name`. So after we get `data` from `api.rooms.show` we need to just do a quick non null assertion, something like this:
```
if(!data) {
    return <div>Could not find the room.</div>
}
```

This means that if data is null or undefined, then we return that div with the error message. But if data is not undefined or null, then we guarantee that it is of the type we want - in this case a "room" object, so typescript gives us nice autocomplete when we try to do dot notation.

Reply:
Is it the same if I put:
`data?.name`
?

Reply:
That is an option, but I would steer you towards the `if` check, purely because if you just drop a `?` in, then the user will either see the intended data as expected, but if for some reason data is undefined (maybe the db is broken, or just temporarily not working for some reason), then the user won't see anything, not even the word "undefined". For that reason, I would do the if check so that we can give the user some feedback if something has gone wrong. Otherwise they may just think the data is just taking a long time to load or something.

Reply:
Done.